presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+            Scanner scan = conn.createScanner("accumulo.metadata", auths);
+            scan.fetchColumnFamily(new Text("loc"));
+
+            // Set the scan range to just this table, from the table ID to the default tablet
+            // row, which is the last listed tablet
+            Key defaultTabletRow = new Key(tableId + '<');
+            Key start = new Key(tableId);
+            Key end = defaultTabletRow.followingKey(PartialKey.ROW);
+            scan.setRange(new Range(start, end));
+
+            String location = null;
+            if (key == null) {
+                // if the key is null, then it is -inf, so get first tablet location
+                for (Entry<Key, Value> kvp : scan) {
+                    location = kvp.getValue().toString();
+                    break;
+                }
+            }
+            else {
+                // Else, we will need to scan through the tablet location data and find the location
+
+                // Create some text objects to do comparison for what we are looking for
+                Text splitCompareKey = new Text();
+                key.getRow(splitCompareKey);
+                Text scannedCompareKey = new Text();
+
+                // Scan the table!
+                for (Entry<Key, Value> kvp : scan) {
+                    // Get the bytes of the key
+                    byte[] keyBytes = kvp.getKey().getRow().copyBytes();
+
+                    // If the last byte is <, then we have hit the default tablet, so use this location
+                    if (keyBytes[keyBytes.length - 1] == '<') {
+                        location = kvp.getValue().toString();
+                        break;
+                    }
+                    else {
+                        // Chop off some magic nonsense
+                        scannedCompareKey.set(keyBytes, 3, keyBytes.length - 3);
+
+                        // Compare the keys, moving along the tablets until the location is found
+                        if (scannedCompareKey.getLength() > 0) {
+                            int compareTo = splitCompareKey.compareTo(scannedCompareKey);
+                            if (compareTo <= 0) {
+                                location = kvp.getValue().toString();
+                            }
+                            else {
+                                // all future tablets will be greater than this key
+                                break;
+                            }
+                        }
+                    }
+                }
+                scan.close();
+            }
+
+            // If we were unable to find the location for some reason, return the default tablet
+            // location
+            return location != null ? location : getDefaultTabletLocation(table);
+        }
+        catch (Exception e) {
+            LOG.error("Failed to get tablet location, returning dummy location", e);
+            e.printStackTrace();
```

Comment:
So please at least put a comment here, to be clear that you purposefully swallow this exception.